src/main/java/com/aws/greengrass/util/platforms/windows/WindowsPlatform.java

```diff
@@ -121,9 +158,100 @@ public void addUserToGroup(String user, String group) throws IOException {
     }
 
     @Override
-    protected void setPermissions(FileSystemPermission permission, Path path,
-                                  EnumSet<FileSystemPermission.Option> options) throws IOException {
-        // [P41372857]: Implement using ACL for Windows
+    protected void setOwner(FileSystemPermission permission, Path path) throws IOException {
+        UserPrincipalLookupService lookupService = path.getFileSystem().getUserPrincipalLookupService();
+        UserPrincipal userPrincipal = lookupService.lookupPrincipalByName(permission.getOwnerUser());
+        FileOwnerAttributeView view = Files.getFileAttributeView(path, FileOwnerAttributeView.class,
+                LinkOption.NOFOLLOW_LINKS);
+
+        logger.atTrace().setEventType(SET_PERMISSIONS_EVENT).kv(PATH, path).kv("owner", permission.getOwnerUser())
+                .log();
+        view.setOwner(userPrincipal);
+
+        // Note that group ownership is not used.
+    }
+
+    @Override
+    protected void setMode(FileSystemPermission permission, Path path) throws IOException {
+        List<AclEntry> acl = aclEntries(permission, path);
+        logger.atTrace().setEventType(SET_PERMISSIONS_EVENT).kv(PATH, path).kv("perm", acl.toString()).log();
+        AclFileAttributeView view = Files.getFileAttributeView(path, AclFileAttributeView.class,
+                LinkOption.NOFOLLOW_LINKS);
+        view.setAcl(acl); // This also clears existing acl!
+    }
+
+    /**
+     * Convert to a list of Acl entries for use with AclFileAttributeView.setAcl.
+     *
+     * @param permission permission to convert
+     * @param path path to apply to
+     * @return List of Acl entries
+     * @throws IOException if any exception occurs while converting to Acl
+     */
+    public static List<AclEntry> aclEntries(FileSystemPermission permission, Path path) throws IOException {
+        UserPrincipalLookupService userPrincipalLookupService = path.getFileSystem().getUserPrincipalLookupService();
+
+        UserPrincipal ownerPrincipal;
+        if (permission.getOwnerUser() == null) {
+            // On Linux, when we set the file permission for the owner, it applies to the current owner and we don't
+            // need to know who the actual owner is. But on Windows, Acl must be associated with an owner.
+            AclFileAttributeView view = Files.getFileAttributeView(path, AclFileAttributeView.class,
+                    LinkOption.NOFOLLOW_LINKS);
+            ownerPrincipal = view.getOwner();
+        } else {
+            ownerPrincipal = userPrincipalLookupService.lookupPrincipalByName(permission.getOwnerUser());
```

Comment:
`IOException` will be thrown. It is a checked exception. It will bubble up to a higher level to handle. For example, [here](https://github.com/aws-greengrass/aws-greengrass-nucleus/blob/main/src/main/java/com/aws/greengrass/lifecyclemanager/KernelAlternatives.java#L150-L151).